### PR TITLE
[autoscaler] rsync cluster

### PR DESCRIPTION
**Pull request by richardliaw**
## What do these changes do?

If `source` and `target` are not provided for `rsync`, `ray rsync-up/rsync-down`
will automatically upload all file_mounts listed in the cluster yaml.

This PR is based off #4782.

## Linter

- [x] I've run `scripts/format.sh` to lint the changes in this PR.

### CODE REVIEW

**Comment by richardliaw** [comment]
This would make https://github.com/ray-project/ray/issues/4004 easier to do.

**Comment by hartikainen** [comment]
> This would make #4004 easier to do.

I don't see the connection between these two. Aren't these two kinda orthogonal?

**Comment by richardliaw** [comment]
> > This would make #4004 easier to do.
> 
> I don't see the connection between these two. Aren't these two kinda orthogonal?

It's indirectly related as this now makes it easier to implement, for both
`rsync` and `up` endpoints.

File: python/ray/autoscaler/commands.py [hunk 1]
```diff
@@ -448,7 +448,12 @@ def rsync(config_file, source, target, override_cluster_name, down):
             rsync = updater.rsync_down
         else:
             rsync = updater.rsync_up
-        rsync(source, target, check_error=False)
+
+        if source and target:
+            rsync(source, target, check_error=False)
+        else:
```
Comment (ericl) [inline python/ray/autoscaler/commands.py]: Shouldn't it throw an error if only 1 of source and target is specified?

**Review by ericl — APPROVED** [review]

File: python/ray/autoscaler/commands.py (hunk 1 above)
Comment (richardliaw) [reply python/ray/autoscaler/commands.py]: OK, cleaned it up.

**Comment by richardliaw** [comment]
(Tested locally and works fine).

**Pull request merged**